src/network/networkd-setlink.c

```diff
@@ -483,6 +515,17 @@ static bool link_is_ready_to_call_set_link(Request *req) {
                         }
                 }
                 break;
+        case SET_LINK_MAC:
+                if (req->netlink_handler == link_set_mac_handler) {
+                        /* This is the second trial to set MTU. Maybe, the first trial failed as the
+                         * interface was up. */
```

Comment:
maybe explicitly say here `On the first attempt req->netlink_handler points to link_set_mac_allow_retry_handler()`, just to make clear why this check works